src/internal/connector/graph/retry_middleware.go

```diff
@@ -50,7 +51,11 @@ func (middleware RetryHandler) retryRequest(
 
 		response, err := pipeline.Next(req, middlewareIndex)
 		if err != nil && !IsErrTimeout(err) {
-			return response, support.ConnectorStackErrorTraceWrap(err, "maximum retries or unretryable")
+			return response,
+				support.ConnectorStackErrorTraceWrap(
+					err,
+					fmt.Sprintf("unretryable error after %d retries", executionCount),
```

Comment:
I was reviewing an error message the other day and realized that this message will get injected, unhelpfully, within many different error strings.  I think we'll end with better error messages by not wrapping a message here at all.  Otherwise we end up with nearly every error saying, `foo: bar: baz: unretryable or failed: qux`, where `qux` normally contains details about the rest failure already.
```suggestion
				clues.Stack(err).WithClues(ctx).With("retry_count", executionCount).WithAll(graph.ErrData(err)...)
```

Reply:
This makes sense, I wanted just expose the retry count was unnecessarily convoluting things.